### PR TITLE
Qiskit new packages

### PR DESCRIPTION
**Pull request by gaboroszkar**
These commits add the core packages, and necessary dependencies for Qiskit, an open-source SDK for working with quantum computers.

I'm happy to be the maintainer of qiskit-terra, qiskit-aer, tweedledum. If needed, I could maintain other packages as well, but with lower priority. I already added myself as maintainer for these packages in the metadata.xml file.

I suggest adding this package to dev-python instead of sci-libs/sci-physics/sci-mathematics, because quantum computers can be regarded as general purpose computers, like current classical computers. This library is made to use quantum computers to compute, not to investigate a specific scientific idea. Current computers also use physics and scientific knowledge to operate, quantum computers just use different set of scientific knowledge for the same purpose, but supposedly, with better algorithmic complexity.

I'm planning on adding more packages, and one meta package for Qiskit, which would install all components of Qiskit. So far, all these packages are enough for the vast majority of users, the most important features work, namely, programming quantum circuits, simulating them with Aer and visualizing the circuits and results. It is jupyter compatible as far as I tested. The test feature also work for these packages on my system. I'd be also happy to maintain other Qiskit components, when they are done.

The corresponding tickets are the following.
https://bugs.gentoo.org/834766
https://bugs.gentoo.org/834767
https://bugs.gentoo.org/834768
https://bugs.gentoo.org/834769
https://bugs.gentoo.org/834770
https://bugs.gentoo.org/834771
https://bugs.gentoo.org/834772

### CODE REVIEW

File: dev-python/qiskit-terra/qiskit-terra-0.19.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,75 @@
+# Copyright 2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DISTUTILS_USE_PEP517=setuptools
+PYTHON_COMPAT=( python3_{7..10} pypy3 )
```
Comment (AndrewAmmerlaan) [inline dev-python/qiskit-terra/qiskit-terra-0.19.2.ebuild:7]:
```suggestion
PYTHON_COMPAT=( python3_{8..10} pypy3 )
```

File: dev-python/qiskit-terra/qiskit-terra-0.19.2.ebuild [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+inherit distutils-r1
+
+DESCRIPTION="Terra is the foundation on which Qiskit is built"
+HOMEPAGE="https://github.com/Qiskit/qiskit-terra"
+SRC_URI="https://github.com/Qiskit/qiskit-terra/archive/refs/tags/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+IUSE="+visualization test"
+RESTRICT="!test? ( test )"
```
Comment (AndrewAmmerlaan) [inline dev-python/qiskit-terra/qiskit-terra-0.19.2.ebuild:18]:
```suggestion
IUSE="+visualization"
```

File: dev-python/qiskit-terra/qiskit-terra-0.19.2.ebuild [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	>=dev-python/tweedledum-1.1
+	visualization? (
+		>=dev-python/matplotlib-3.3
+		>=dev-python/ipywidgets-7.3.0
+		dev-python/pydot
+		>=dev-python/pillow-4.2.1
+		>=dev-python/pylatexenc-1.4
+		>=dev-python/seaborn-0.9.0
+		>=dev-python/pygments-2.4
+	)"
```
Comment (AndrewAmmerlaan) [inline dev-python/qiskit-terra/qiskit-terra-0.19.2.ebuild:51]: Missing `[${PYTHON_USEDEP}]`?

File: dev-python/qiskit-terra/qiskit-terra-0.19.2.ebuild [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+distutils_enable_tests pytest
+
+python_prepare_all() {
+	# Small issues with the tests.
+	# qiskit.Aer module depends on qiskit-terra, it cannot be used,
+	# and an exact comparison of float switched to approximate comparison.
+	eapply "${FILESDIR}/test_corrections.patch"
```
Comment (AndrewAmmerlaan) [inline dev-python/qiskit-terra/qiskit-terra-0.19.2.ebuild:59]: Same as above here

File: dev-python/retworkx/metadata.xml [hunk 5]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "https://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+  <!-- comaintainers-welcomed -->
+  <maintainer type="person">
```
Comment (AndrewAmmerlaan) [inline dev-python/retworkx/metadata.xml:5]: Same as above here

File: dev-python/retworkx/retworkx-0.11.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,109 @@
+# Copyright 2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DISTUTILS_USE_PEP517=setuptools
+PYTHON_COMPAT=( python3_{7..10} pypy3 )
```
Comment (AndrewAmmerlaan) [inline dev-python/retworkx/retworkx-0.11.0.ebuild:7]:
```suggestion
PYTHON_COMPAT=( python3_{8..10} pypy3 )
```

File: dev-python/retworkx/retworkx-0.11.0.ebuild [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+		dev-python/fixtures
+		>=dev-python/testtools-2.5.0
+		>=dev-python/networkx-2.5
+		dev-python/stestr
+		media-gfx/graphviz[gts]
+		dev-python/graphviz
+	)"
+
+RDEPEND=">=dev-python/numpy-1.16.0"
+
```
Comment (AndrewAmmerlaan) [inline dev-python/retworkx/retworkx-0.11.0.ebuild:101]: `distutils_enable_tests pytest` appears to be missing here. Please add it (and drop IUSE and RESTRICT)

File: dev-python/retworkx/retworkx-0.11.0.ebuild [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+SLOT="0"
+IUSE="test"
+RESTRICT="!test? ( test )"
+KEYWORDS="~amd64"
+
+BDEPEND="dev-python/setuptools_rust
+	test? (
+		dev-python/fixtures
+		>=dev-python/testtools-2.5.0
+		>=dev-python/networkx-2.5
```
Comment (AndrewAmmerlaan) [inline dev-python/retworkx/retworkx-0.11.0.ebuild:94]: Missing ` [${PYTHON_USEDEP}]`?

File: dev-python/tweedledum/metadata.xml [hunk 9]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "https://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+  <!-- comaintainers-welcomed -->
+  <maintainer type="person">
```
Comment (AndrewAmmerlaan) [inline dev-python/tweedledum/metadata.xml:5]: Same as above here

File: dev-python/tweedledum/tweedledum-1.1.1.ebuild [hunk 10]
```diff
@@ -0,0 +1,33 @@
+# Copyright 2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DISTUTILS_USE_PEP517=setuptools
+PYTHON_COMPAT=( python3_{6..10} pypy3 )
```
Comment (AndrewAmmerlaan) [inline dev-python/tweedledum/tweedledum-1.1.1.ebuild:7]:
```suggestion
PYTHON_COMPAT=( python3_{8..10} pypy3 )
```

File: dev-python/tweedledum/tweedledum-1.1.1.ebuild [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+
+inherit distutils-r1
+
+DESCRIPTION="Library for analysis, compilation, synthesis, optimization of quantum circuits"
+HOMEPAGE="https://github.com/boschmitt/tweedledum"
+SRC_URI="https://github.com/boschmitt/tweedledum/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+RESTRICT="!test? ( test )"
```
Comment (AndrewAmmerlaan) [inline dev-python/tweedledum/tweedledum-1.1.1.ebuild:17]:
```suggestion
```

**Comment by gaboroszkar** [comment]
@AndrewAmmerlaan thanks for review and suggestions, I resolved them.

**Comment by AndrewAmmerlaan** [comment]
> @AndrewAmmerlaan thanks for review and suggestions, I resolved them.

Awesome :+1: 

I reminded mgorny of your ::guru access request. He has added you and you should be able to push there now. Let's add this first to ::guru so the [Tinderbox](https://blogs.gentoo.org/ago/2020/07/04/gentoo-tinderbox/) can do its thing, and than later maybe move this to ::gentoo.

**Comment by gaboroszkar** [comment]
These packages are added to ::guru.
The proxy information is missing from the packages in ::guru, but they should be included in the packages in ::gentoo.

**Review by AndrewAmmerlaan — APPROVED** [review]

**Comment by tamiko** [comment]
@gaboroszkar Thank you for the packaging. I will try to have a look at this pull request sometime in the next two weeks. What about we consider adding the packages as port of the science team but add you as a proxy maintainer as well?

**Comment by thesamesam** [comment]
Graet work! I've added github & pypi remote-ids and made some minor style changes too.

Welcome!

**Comment by thesamesam** [comment]
ping wrt https://bugs.gentoo.org/865129.

**Pull request closed without merging**